src/ng/directive/input.js

```diff
@@ -1248,6 +1248,14 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
       composing = true;
     });
 
+    // Support: IE9+
+    element.on('compositionupdate', function(ev) {
+      // End composition when ev.data is empty string on 'compositionupdate' event.
+      // When the input de-focusses (e.g. by clicking away), IE triggers 'compositionupdate'
+      // instead of 'compositionend'.
+      if (ev.data === '') composing = false;
```

Comment:
Setting `composing` to `false` isn't enough. You also need to call the `listener`.
Maybe add a check here and in `compositionend` listener below, to only call `listener()` if `composing` was `true` before (to avoid calling it twice on browsers that may trigger both events - or is that impossible?)

Reply:
Well, I've tested it in a real world scenario and it is enough, because IE also fires the change event afterwards.
We could still implement your suggestion. It might help with another IE issue: https://github.com/angular/angular.js/issues/13617

Reply:
I see. Makes sense. Any idea why tests fail on Travis then?